app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -327,92 +323,10 @@ class _CustomChatInputState extends ConsumerState<CustomChatInput> {
-                            onTapImage: () async {
-                              XFile? imageFile = await ImagePicker()
-                                  .pickImage(source: ImageSource.gallery);
-                              if (imageFile != null) {
-                                List<File> files = [File(imageFile.path)];
-                                final mimeType =
-                                    lookupMimeType(files.first.path);
-                                if (!context.mounted) {
-                                  return;
-                                }
-                                attachmentConfirmationDialog(
-                                  context,
-                                  files,
-                                  _FileWidget(mimeType, files.first),
-                                  handleFileUpload,
-                                  AttachmentType.image,
-                                );
-                              }
-                            },
-                            onTapVideo: () async {
-                              XFile? imageFile = await ImagePicker()
-                                  .pickVideo(source: ImageSource.gallery);
-                              if (imageFile != null) {
-                                List<File> files = [File(imageFile.path)];
-                                final mimeType =
-                                    lookupMimeType(files.first.path);
-                                if (!context.mounted) {
-                                  return;
-                                }
-                                attachmentConfirmationDialog(
-                                  context,
-                                  files,
-                                  _FileWidget(mimeType, files.first),
-                                  handleFileUpload,
-                                  AttachmentType.video,
-                                );
-                              }
-                            },
-                            onTapFile: () async {
-                              var selectedFiles = await handleFileSelection(
-                                ctx,
-                              );
-                              final mimeType =
-                                  lookupMimeType(selectedFiles!.first.path);
-                              if (!context.mounted) {
-                                return;
-                              }
-                              attachmentConfirmationDialog(
-                                context,
-                                selectedFiles,
-                                _FileWidget(mimeType, selectedFiles.first),
-                                handleFileUpload,
-                                AttachmentType.file,
-                              );
-                            },
-                          ),
+                        onTap: () => showAttachmentSelection(
```

Comment:
can you elaborate why you felt the need to combine the attachment management for objects with this for chat? The `null` here in there kinda indicates pretty clearly to me that these are distinct enough to not combine them ... or is there any other place we send this over _without_ a manager attached?

or, alternatively, if you feel strong about them being the same components, how about not making the `showModalBottomSheet` any `manager` at all and instead manage this in a different layer above?

Reply:
They aren't same in a way that chat attachments don't need manager in any case, while the generic attachment do require a manager. My probable opinion to have this improved is to think it as different from ```manager``` since these attachments flow end up on ```timelineStream```. Shall I revert this specific change and keep the old flow?

Reply:
Yeah, I'd rather we do not touch the existing and working chat-attachments-code in this PR but keep it the way it is. We can still think about unifying that later...